.psalm/psalm-loader.php

```diff
@@ -5,10 +5,63 @@
  * @package wp-job-manager
  */
 
-if ( ! defined( 'ABSPATH' ) ) {
-	// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedConstantFound
-	define( 'ABSPATH', __DIR__ . '/../' );
-}
+/** WordPress Constants */
+// phpcs:disable WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedConstantFound
+define( 'ABSPATH', __DIR__ . '/../' );
+
+// ./wp-includes/default-constants.php
+
+define( 'WP_DEBUG', true );
+define( 'WP_DEBUG_LOG', true );
```

Comment:
Is there an alternative to having the copy of all the WordPress constants? It doesn't seem maintainable. Maybe https://github.com/psalm/psalm-plugin-wordpress has better stubs?

Reply:
This is basically what that plugin does: https://github.com/psalm/psalm-plugin-wordpress/blob/master/stubs/globals.php

However, I'll remove the constants from the loader and just use that plugin directly. :)

Reply:
I checked here and, as mentioned on Slack, that plugin tries to parse all the `add_filter` / `add_action` calls and:

- Throws parse errors when it finds a hook name using a class constant
- When the above parse errors are fixed, it throws errors about not knowing about WPJM custom hooks
- The above error can't be fixed easily because https://github.com/wp-hooks/generator - which is supposed to generate a file containing all the WPJM hooks - doesn't work (it tries to load a class that doesn't exist)

So for now we have two options:

1. Just keep the WP constants there 
2. Or, install the plugin but do not enable it on psalm. Instead, load the file referred above directly here in the loader.

WDYT?

Reply:
I'd probably defer to the psalm-plugin-wordpress plugin for the constant list and include it from there, but I imagine it won't change that often so no strong opinions on it.

Reply:
Let's go this way for now, in any case we can change in the future. :)